### PR TITLE
CORE-9385 Refactoring ledger transaction verification

### PR DESCRIPTION
**Pull request by mbrkic-r3**
- Ledger transaction verification moved to lib "ledger-utxo-transaction-verifier".
- Refactored UtxoLedgerTransactionVerifierService that verifies ledger transaction using external events to send requests via message bus to Verification Processor.
- Updated VerificationRequestHandlerImpl to perform all transaction verifications (not just contract verification).
- Refactored existing code to use UtxoLedgerTransactionVerifierService.

### CODE REVIEW

File: components/ledger/ledger-common-flow/src/main/kotlin/net/corda/ledger/common/flow/impl/transaction/factory/TransactionMetadataFactoryImpl.kt [hunk 1]
```diff
@@ -48,8 +48,8 @@ class TransactionMetadataFactoryImpl @Activate constructor(
             CordaPackageSummaryImpl(
                 name = cpk.cpkId.name,
                 version = cpk.cpkId.version,
-                signerSummaryHash = cpk.cpkId.signerSummaryHash.toHexString(),
-                fileChecksum = cpk.fileChecksum.toHexString()
```
Comment (mbrkic-r3) [inline components/ledger/ledger-common-flow/src/main/kotlin/net/corda/ledger/common/flow/impl/transaction/factory/TransactionMetadataFactoryImpl.kt:52]: Changed so it's consistent with other code and that SecureHash could be created from String representation.

File: components/ledger/ledger-consensual-flow/src/main/kotlin/net/corda/ledger/consensual/flow/impl/transaction/serializer/amqp/ConsensualSignedTransactionSerializer.kt [hunk 2]
```diff
@@ -19,7 +18,7 @@ import org.osgi.service.component.annotations.Reference
 import org.osgi.service.component.annotations.ServiceScope.PROTOTYPE
 
 @Component(
-    service = [InternalCustomSerializer::class, UsedByFlow::class, UsedByVerification::class],
```
Comment (mbrkic-r3) [inline components/ledger/ledger-consensual-flow/src/main/kotlin/net/corda/ledger/consensual/flow/impl/transaction/serializer/amqp/ConsensualSignedTransactionSerializer.kt:22]: Not needed in verification sandbox and it's was causing issues as it's dependencies were not available.

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/serializer/amqp/UtxoSignedTransactionSerializer.kt [hunk 3]
```diff
@@ -20,7 +19,7 @@ import org.osgi.service.component.annotations.Reference
 import org.osgi.service.component.annotations.ServiceScope.PROTOTYPE
 
 @Component(
-    service = [ InternalCustomSerializer::class, UsedByFlow::class, UsedByVerification::class ],
```
Comment (mbrkic-r3) [inline components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/serializer/amqp/UtxoSignedTransactionSerializer.kt:23]: Not needed in verification sandbox and it's was causing issues as it's dependencies were not available.

File: testing/cpbs/ledger-utxo-demo-app/src/main/kotlin/net/cordapp/demo/utxo/UtxoDemoFlow.kt [hunk 4]
```diff
@@ -31,8 +31,6 @@ import java.time.Instant
 class UtxoDemoFlow : ClientStartableFlow {
     data class InputMessage(val input: String, val members: List<String>, val notary: String)
 
-    class TestCommand : Command
```
Comment (mbrkic-r3) [inline testing/cpbs/ledger-utxo-demo-app/src/main/kotlin/net/cordapp/demo/utxo/UtxoDemoFlow.kt:34]: Moved to contract CPK so it's visible in Verification sandbox

File: components/ledger/ledger-verification/build.gradle [hunk 5]
```diff
@@ -20,6 +20,15 @@ dependencies {
 
     implementation platform("net.corda:corda-api:$cordaApiVersion")
 
+    implementation 'net.corda:corda-config-schema'
+    implementation 'net.corda:corda-db-schema'
+    implementation 'net.corda:corda-ledger-common'
+    implementation 'net.corda:corda-ledger-consensual'
```
Comment (vlajos) [inline components/ledger/ledger-verification/build.gradle]: Why is this consensual dependency needed?

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoLedgerTransactionVerifierService.kt [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package net.corda.ledger.utxo.flow.impl.transaction.verifier
+
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.ledger.utxo.transaction.UtxoLedgerTransaction
+
+/**
+ * [UtxoLedgerTransactionVerifierService] verifies UTXO ledger transactions.
+ */
+interface UtxoLedgerTransactionVerifierService {
```
Comment (lankydan) [inline components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoLedgerTransactionVerifierService.kt]: Can we call this `UtxoLedgerTransactionVerifier` or `UtxoLedgerTransactionVerificationService`.

**Review by lankydan — COMMENTED** [review]
One, minor comment otherwise looks ok.

File: components/ledger/ledger-verification/build.gradle (hunk 5 above)
Comment (mbrkic-r3) [reply components/ledger/ledger-verification/build.gradle]: Removed

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoLedgerTransactionVerifierService.kt (hunk 6 above)
Comment (mbrkic-r3) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoLedgerTransactionVerifierService.kt]: Renamed

File: libs/ledger/ledger-utxo-data/src/main/kotlin/net/corda/ledger/utxo/data/state/serializer/amqp/StateAndRefSerializer.kt [hunk 7]
```diff
@@ -0,0 +1,73 @@
+package net.corda.ledger.utxo.data.state.serializer.amqp
```
Comment (vlajos) [inline libs/ledger/ledger-utxo-data/src/main/kotlin/net/corda/ledger/utxo/data/state/serializer/amqp/StateAndRefSerializer.kt:1]: Would not be the ledger-utxo-flow component a better place for these serializers?

Comment (mbrkic-r3) [reply libs/ledger/ledger-utxo-data/src/main/kotlin/net/corda/ledger/utxo/data/state/serializer/amqp/StateAndRefSerializer.kt:1]: I wouldn't like to have dependency on `*-flow` in Verification processor.

**Review by vlajos — DISMISSED** [review]
LGTM

**Comment by mbrkic-r3** [comment]
I'm going to merge this PR, if there will be any additional comments and need to change something, I'll address it with a new PR

File: libs/ledger/ledger-utxo-data/src/main/kotlin/net/corda/ledger/utxo/data/state/serializer/amqp/StateAndRefSerializer.kt (hunk 7 above)
Comment (blsemo) [reply libs/ledger/ledger-utxo-data/src/main/kotlin/net/corda/ledger/utxo/data/state/serializer/amqp/StateAndRefSerializer.kt:1]: Should be ok to have it here. Serializers are allowed in the data module.

**Review by blsemo — APPROVED** [review]
Looks ok to me.

**Review by mcgovc — APPROVED** [review]
LGTM - build changes

**Pull request merged**